node/src/BaseClient.ts

```diff
@@ -498,7 +498,10 @@ export type ReadFrom =
     | "primary"
     /** Spread the requests between all replicas in a round robin manner.
         If no replica is available, route the requests to the primary.*/
-    | "preferReplica";
+    | "preferReplica"
+    /** Spread the requests between replicas in the same client's AZ (Availability zone) in a round robin manner.
+        If no replica is available, route the requests to the primary.*/
+    | "AZAffinity"
```

Comment:
```suggestion
    | { type: "AZAffinity", clientAz?: string; }
```
That is more correct location of `clientAz`.

Reply:
I think it's possible that in the future we will want to support fully baked support for EC2, EKS and all the known and widely used platform. Meaning, the user will provide something else than `clientAz`, and I'm afraid that putting those together will make issues later.
Try to think about the possible design and let me know if you don't think it will affect. IDK, the designs options doc is existing, but no decision has been taken, so I can speculate as you. I can share the doc if you want to dive into this question more deeply.